### PR TITLE
Precipitation Widget HTML & Logic & Tests, Search Bar Colours, Page Toggle Sticky

### LINKED ISSUE #56: precipitation widget html
Create precipitation rain cloud with animated rain drops using html, css, and js.

### LINKED ISSUE #129: Make page toggle button sticky to bottom of screen
Make page switching button fixed to the bottom right side of the screen with an appropriate z-index so it appears on top of everything else and is accessible.

### LINKED ISSUE #146: Change colors of search bar to fit more in tune with the rest of the dashboard
Title is self-explanatory.

### LINKED ISSUE #161: precipitation widget logic
## In this issue:

- Change amount of rain in animation depending on precipitation amount.

- Change the frequency of the cloud's animation depending on precipitation amount as well.

### LINKED ISSUE #185: Create tests for precipitation widget
Title is self-explanatory.

### PR DESCRIPTION
**Pull request by abijeet-dhillon**
## In this PR:

- Created a precipitation widget which is an animation of a rain cloud. The amount of rain falling is dependent on the precipitation amount (see precipitation.js for the scale). Also wrote tests for little testable code there was written for this widget.

- Changed the colours of the search bar to fit the dark mode theme better on both the fire and weather pages.

- Page toggle button sticks to the bottom right of the pages.

There was a previous pull request for the only the precipitation html, search bar colours, and page toggle but I seemed to have corrupted that branch...😅 Disregard that previous pull request, this one includes all of the code from that one and more!🤩

Closes #56 #146 #129 #161 #185

### CODE REVIEW

File: Site/bc_weather_station_dashboard/website/static/JavaScript/precipitation.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        rainAmount = 100;
+    } else if(precipitationValue <= 2.0) {
+        rainAmount = 60;
+    } else if(precipitationValue <= 4.0) {
+        rainAmount = 40;
+    } else if(precipitationValue <= 6.0) {
+        rainAmount = 20;
+    } else {
+        rainAmount = 5;
+    }
```
Comment (cmcahoon01) [inline Site/bc_weather_station_dashboard/website/static/JavaScript/precipitation.js]: Try replacing this big block with this:
`var rainAmount = 100/(precipitationValue+.1);`
It should give you roughly the same values

Comment (abijeet-dhillon) [reply Site/bc_weather_station_dashboard/website/static/JavaScript/precipitation.js]: Changed it, good idea

**Review by robertyac — APPROVED** [review]
Nice I approved your corrupted branch if i recall. good work

**Review by Davis245 — APPROVED** [review]
Looks good. Well done.

**Pull request merged**